<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [cxx-abi-dev] __cxa_vec_new2 and deallocation functions that	throw exceptions
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:cxx-abi-dev%40codesourcery.com?Subject=Re%3A%20%5Bcxx-abi-dev%5D%20__cxa_vec_new2%20and%20deallocation%20functions%20that%0A%09throw%20exceptions&In-Reply-To=%3C1052984298.29358.2.camel%40minax.codesourcery.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="001715.html">
   <LINK REL="Next"  HREF="001717.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[cxx-abi-dev] __cxa_vec_new2 and deallocation functions that	throw exceptions</H1>
    <B>Mark Mitchell</B> 
    <A HREF="mailto:cxx-abi-dev%40codesourcery.com?Subject=Re%3A%20%5Bcxx-abi-dev%5D%20__cxa_vec_new2%20and%20deallocation%20functions%20that%0A%09throw%20exceptions&In-Reply-To=%3C1052984298.29358.2.camel%40minax.codesourcery.com%3E"
       TITLE="[cxx-abi-dev] __cxa_vec_new2 and deallocation functions that	throw exceptions">mark at codesourcery.com
       </A><BR>
    <I>Thu May 15 07:38:19 UTC 2003</I>
    <P><UL>
        <LI>Previous message: <A HREF="001715.html">[cxx-abi-dev] __cxa_vec_new2 and deallocation functions that throw exceptions
</A></li>
        <LI>Next message: <A HREF="001717.html">[cxx-abi-dev] __cxa_vec_new2 and deallocation functions that throw exceptions
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1716">[ date ]</a>
              <a href="thread.html#1716">[ thread ]</a>
              <a href="subject.html#1716">[ subject ]</a>
              <a href="author.html#1716">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>On Thu, 2003-05-15 at 00:13, Dennis Handly wrote:
&gt;<i> &gt;[except.terminate]
</I>&gt;<i> &gt;  --when the exception handling mechanism, after completing evaluation
</I>&gt;<i> &gt;    of the expression to be thrown but before the exception is caught
</I>&gt;<i> &gt;    (_except.throw_), calls a user function that exits via an uncaught
</I>&gt;<i> &gt;    exception,1)
</I>&gt;<i> 
</I>&gt;<i> So what does uncaught exception mean?  Any exception without a handler?
</I>&gt;<i> (That footnote needs to be expanded.)
</I>&gt;<i>
</I>&gt;<i> Or anytime you throw when uncaught_exception is true, 18.6.4(2)?
</I>&gt;<i> 
</I>&gt;<i> This disallows throws that are caught locally in destructors, while
</I>&gt;<i> unwinding.  Or is this not &quot;exiting&quot;, so legal?
</I>
A function can exit either by use of &quot;return&quot; (including falling off the
end of the function), by throwing exception, or by calling longjmp.  If
a destructor catches the exception, it does not exit by throwing an
exception.

So, yes, the situation you describe is legal.

-- 
Mark Mitchell &lt;<A HREF="http://sourcerytools.com/cgi-bin/mailman/listinfo/cxx-abi-dev">mark at codesourcery.com</A>&gt;
CodeSourcery, LLC


</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="001715.html">[cxx-abi-dev] __cxa_vec_new2 and deallocation functions that throw exceptions
</A></li>
	<LI>Next message: <A HREF="001717.html">[cxx-abi-dev] __cxa_vec_new2 and deallocation functions that throw exceptions
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1716">[ date ]</a>
              <a href="thread.html#1716">[ thread ]</a>
              <a href="subject.html#1716">[ subject ]</a>
              <a href="author.html#1716">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://sourcerytools.com/cgi-bin/mailman/listinfo/cxx-abi-dev">More information about the cxx-abi-dev
mailing list</a><br>
</body></html>
